src/server.rs

```diff
@@ -166,6 +191,20 @@ impl ServerStats {
             }};
         }
 
+        macro_rules! set_duration_stat {
+            ($vec:ident, $dur:expr, $num:expr, $name:expr) => {{
+                let mut stat = CacheStatistic::new();
+                stat.set_name(String::from($name));
+                if $num > 0 {
+                    let duration = $dur / $num as u32;
+                    stat.set_str(format!("{}.{:03} s", duration.as_secs(), duration.subsec_nanos() / 1000_000));
+                } else {
+                    stat.set_str("0.000 s".to_owned());
```

Comment:
I wonder if there'd be any value in adding a `Duration` case to the `CacheStatistic` enum? We don't currently do anything interesting with it, although I'd like to add a way to have `--show-stats` output a JSON representation of the stats for consumption by other programs. (Granted you're just duplicating what I previously implemented here...)